c7n/resources/asg.py

```diff
@@ -1216,9 +1216,19 @@ def process_asg(self, asg):
             instances = [self.instance_map[i] for i in instance_ids]
             self.prune_instance_tags(client, asg, tag_set, instances)
         if not self.manager.config.dryrun:
-            client.create_tags(
-                Resources=instance_ids,
-                Tags=[{'Key': k, 'Value': v} for k, v in tag_map.items()])
+            while instance_ids:
```

Comment:
this might be nice to extract as a context manager.

Reply:
Should the context manager be inside the except block, like exactly where the extraction operation is happening? Thanks!

Reply:
ideally the context manager encapsulates all of the exception handler ie.

```python
with retry_remaining_instances(instance_ids):
      client.create_tags(Resources=instance_ids, Tags=tags)
```

Reply:
Switched to use context manager